airflow/upgrade/rules/mesos_executor_removed.py

```diff
@@ -0,0 +1,35 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from airflow.upgrade.rules.base_rule import BaseRule
+from airflow.configuration import conf
+
+
+class MesosExecutorRemovedRule(BaseRule):
+    """
+    MesosExecutorRemovedRule class to ease upgrade to Airflow 2.0
+    """
+
+    title = "Removal of Mesos Executor"
+    description = """
+    The Mesos Executor has been deprecated as it was not widely used and not maintained.
+    """
+
+    def check(self):
+        mesos_config = conf.getsection("mesos")
```

Comment:
The existence of this section in the configuration file does not prevent migration to Airflow 2.0.  I think we should check that the `executor` option in the `core` section is not set to `MesosExecutor` and warn the user that he needs to migrate to another supported alternative such as CeleryExecutor.

Reply:
Thanks for the reply @mik-laj I will make the changes.